pkg/collector/corechecks/containers/docker/bundled_events.go

```diff
@@ -0,0 +1,80 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build docker
+// +build docker
+
+package docker
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/docker"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+func newBundledTransformer(hostname string, types []string) eventTransformer {
+	filteredEventTypes := make(map[string]struct{}, len(types))
+	for _, t := range types {
+		filteredEventTypes[t] = struct{}{}
+	}
+
+	return &bundledTransformer{
+		hostname:           hostname,
+		filteredEventTypes: filteredEventTypes,
+	}
+}
+
```

Comment:
maybe we can add a comment that in agent8 we would like to remove the `bundledTransformer`